services/apigatewayv2/pkg/resource/api/custom_http_api.go

```diff
@@ -0,0 +1,413 @@
+
+		return errors.New(errorMessage)
+	} else {
+		// Body field is present.
+		// Check that no other fields except 'Basepath' and 'FailOnWarnings' is present.
+		specCopy := api.Spec.DeepCopy()
+		specCopy.Body = nil
+		specCopy.FailOnWarnings = nil
+		specCopy.Basepath = nil
+		opts := []cmp.Option{cmpopts.EquateEmpty()}
+		if cmp.Equal(*specCopy, v1alpha1.APISpec{}, opts...) {
+			return nil
+		} else {
+			return errors.New("only 'FailOnWarnings' and 'Basepath' fields can be used with 'Body' field")
+		}
+	}
+}
+
+// validateReimportApiInputFields validates if all the fields are present for a successful ReimportApi operation
+// Currently this validation is similar to ImportApi validation.
+func (rm *resourceManager) validateReimportApiInputFields(api *v1alpha1.API) error {
+	return rm.validateImportApiInputFields(api)
+}
+
+// validateCreateApiInputFields validates if all the fields are present for a successful CreateApi operation
+func (rm *resourceManager) validateCreateApiInputFields(api *v1alpha1.API) error {
+	if api.Spec.Name == nil || api.Spec.ProtocolType == nil {
+		return errors.New("'Name' and 'ProtocolType' are required properties if 'Body' field is not present")
+	}
+	return nil
+}
+
+// validateUpdateApiInputFields validates if all the fields are present for a successful UpdateApi operation
+// Currently this validation is similar to CreateApi validation.
+func (rm *resourceManager) validateUpdateApiInputFields(api *v1alpha1.API) error {
+	return rm.validateCreateApiInputFields(api)
+}
+
+// importApi creates the Api resource by performing ImportApi sdk operation
+func (rm *resourceManager) importApi(ctx context.Context, desired *resource) (*resource, error) {
+	input, err := rm.importApiInput(desired)
+	if err != nil {
+		return nil, err
+	}
+
+	resp, respErr := rm.sdkapi.ImportApiWithContext(ctx, input)
+	rm.metrics.RecordAPICall("CREATE", "ImportApi", respErr)
+	if respErr != nil {
+		return nil, respErr
+	}
+
+	// Merge in the information we read from the API call above to the copy of
+	// the original Kubernetes object we passed to the function
+	ko := desired.ko.DeepCopy()
+
+	if resp.ApiEndpoint != nil {
+		ko.Status.APIEndpoint = resp.ApiEndpoint
+	}
+	if resp.ApiGatewayManaged != nil {
+		ko.Status.APIGatewayManaged = resp.ApiGatewayManaged
+	}
+	if resp.ApiId != nil {
+		ko.Status.APIID = resp.ApiId
+	}
+	if resp.CreatedDate != nil {
+		ko.Status.CreatedDate = &metav1.Time{*resp.CreatedDate}
+	}
+	if resp.ImportInfo != nil {
+		f9 := []*string{}
+		for _, f9iter := range resp.ImportInfo {
+			var f9elem string
+			f9elem = *f9iter
+			f9 = append(f9, &f9elem)
+		}
+		ko.Status.ImportInfo = f9
+	}
+	if resp.Warnings != nil {
+		f15 := []*string{}
+		for _, f15iter := range resp.Warnings {
+			var f15elem string
+			f15elem = *f15iter
+			f15 = append(f15, &f15elem)
+		}
+		ko.Status.Warnings = f15
+	}
+
+	rm.setStatusDefaults(ko)
+
+	return &resource{ko}, nil
```

Comment:
I'm working on a way to generate this code instead of needing to maintain it by hand in these custom creation scenarios.